operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResourceConfig.java

```diff
@@ -10,20 +12,32 @@ public class KubernetesDependentResourceConfig {
 
   private Set<String> namespaces = Constants.SAME_AS_CONTROLLER_NAMESPACES_SET;
   private String labelSelector = NO_VALUE_SET;
-
   private boolean namespacesWereConfigured = false;
 
+  @SuppressWarnings("rawtypes")
+  protected Predicate onAddFilter;
```

Comment:
No, the thing is it comes from annotations in this case, can make paramtrized class, at the end itt will not help with that problem, since in the annotation eventually are no generics (KubernetesDependent). But at least on this layer might look better, true.

Reply:
pushed.